exec/plugin-commands-script-runners/src/dlx.ts

```diff
@@ -71,48 +78,59 @@ export async function handler (
   [command, ...args]: string[]
 ): Promise<{ exitCode: number }> {
   const pkgs = opts.package ?? [command]
-  const { cacheLink, cacheExists, cachedDir } = findCache(pkgs, {
-    dlxCacheMaxAge: opts.dlxCacheMaxAge,
-    cacheDir: opts.cacheDir,
-    registries: opts.registries,
-  })
-  if (!cacheExists) {
-    fs.mkdirSync(cachedDir, { recursive: true })
+  let cache: FindCacheResult | undefined
+  let prepareDir: string
+  if (!!opts.ignoreModulesCache || pkgs.some(pkg => pkg.endsWith('@latest'))) {
+    prepareDir = tempy.directory({ prefix: 'pnpm-dlx-' })
+  } else {
+    cache = findCache(pkgs, {
+      dlxCacheMaxAge: opts.dlxCacheMaxAge,
+      cacheDir: opts.cacheDir,
+      registries: opts.registries,
+    })
+    prepareDir = cache.dir
+  }
```

Comment:
You probably could just replace the cacheDir with the temp dir. It will do a few redundant filesytem operations but the difference shouldn't be noticeable.

```suggestion
   let cacheDir: string
   if (!!opts.ignoreModulesCache || pkgs.some(pkg => pkg.endsWith('@latest'))) {
     cacheDir = tempy.directory({ prefix: 'pnpm-dlx-' })
   } else {
     cacheDir = opts.cacheDir
   }
   const { cacheLink, cacheExists, cachedDir } = findCache(pkgs, {
     dlxCacheMaxAge: opts.dlxCacheMaxAge,
     cacheDir,
     registries: opts.registries,
   })
   if (!cacheExists) {
     fs.mkdirSync(cachedDir, { recursive: true })
```

Reply:
https://github.com/pnpm/pnpm/pull/8722/commits/5c3021e5b849723670a162b2e131729e4c89491d

(I decided to keep `cache.dir` instead of `cachedDir` because `cachedDir` and `cacheDir` look too similar to each other, which may cause confusion)